improve description for Swift Package Manager

sahara-ooga opened the pull request:
Change description for Swift Package Manager in README.md.
Thanks for review!

I refer to README.md in [apple/swift-argument-parser](https://github.com/apple/swift-argument-parser) in writing it.

sahara-ooga committed:
improve description for Swift Package Manager

sahara-ooga committed:
add Target description

sahara-ooga commented on README.md:
````diff
@@ -293,11 +293,31 @@ github "YusukeHosonuma/SwiftPrettyPrint"
 ```
 
-or add from Xcode 10+.
+Finally, include "SwiftPrettyPrint" as a dependency for your executable target:
+
+```swift
+let package = Package(
+    // name, platforms, products, etc.
+    dependencies: [
+        .package(url: "https://github.com/YusukeHosonuma/SwiftPrettyPrint.git", .upToNextMajor(from: "1.0.0")),
+        // other dependencies
+    ],
+    targets: [
+        .target(name: "<your-target-name>", dependencies: [
+            .product(name: "SwiftPrettyPrint", package: "SwiftPrettyPrint"),
+        ]),
+        // other targets
+    ]
+)
````
This target description is needed in fact.

YusukeHosonuma commented on README.md:
````diff
@@ -293,11 +293,31 @@ github "YusukeHosonuma/SwiftPrettyPrint"
 ```
 
-or add from Xcode 10+.
+Finally, include "SwiftPrettyPrint" as a dependency for your executable target:
````
This library is not locked in to executable target.

```suggestion
Finally, include "SwiftPrettyPrint" as a dependency for your any target:
```

YusukeHosonuma commented on README.md:
```diff
@@ -293,11 +293,31 @@ github "YusukeHosonuma/SwiftPrettyPrint"
+        // other dependencies
+    ],
+    targets: [
+        .target(name: "<your-target-name>", dependencies: [
+            .product(name: "SwiftPrettyPrint", package: "SwiftPrettyPrint"),
+        ]),
```
The `dependencies` can described by name when package name and library name are same.

```suggestion
        .target(name: "<your-target-name>", dependencies: ["SwiftPrettyPrint"]),
```

YusukeHosonuma reviewed (commented):
Good improvements, please confirm to comment.

sahara-ooga committed:
Update README.md

This library is not locked in to executable target.

Co-authored-by: Yusuke Hosonuma <tobi462@gmail.com>

sahara-ooga committed:
Update README.md

The dependencies can described by name when package name and library name are same.

Co-authored-by: Yusuke Hosonuma <tobi462@gmail.com>

sahara-ooga commented:
I imported suggestions ! I appriciate your review & comments!
